code/game/machinery/mind_machine.dm

```diff
@@ -0,0 +1,846 @@
+	var/occupantScan = FALSE
+	var/manipRating = 4 //Determines speed of swap/malfunction chance on delay swap
+	var/scanRatingOne = 6 //Determines if you can swap silicons
+	var/scanRatingTwo = 6
+	var/errorMessage = "No errors"
+	var/badSwap = FALSE
+	var/malfSwap = FALSE //Triggers malfunction if true
+	var/list/illegalSwap = list() //Might break the game/remove the player from the round. If you get mechahitler in there you deserve your round, though
+
+//////Parts and connection/////////
+
+/obj/machinery/mind_machine/mind_machine_hub/New()
+	illegalSwap = boss_mobs + blacklisted_mobs - list(/mob/living/simple_animal/hostile/mechahitler, /mob/living/simple_animal/hostile/alien/queen/large, /mob/living/simple_animal/hostile/retaliate/cockatrice)
+	component_parts = newlist(
+		/obj/item/weapon/circuitboard/mind_machine_hub,
+		/obj/item/weapon/stock_parts/console_screen,
+		/obj/item/weapon/stock_parts/manipulator/nano,
+		/obj/item/weapon/stock_parts/manipulator/nano,
+		/obj/item/weapon/stock_parts/subspace/analyzer,
+		/obj/item/weapon/stock_parts/subspace/filter,
+		/obj/item/weapon/stock_parts/subspace/amplifier
+	)
+
+/obj/machinery/mind_machine/mind_machine_hub/RefreshParts()
+	var/manipCount = 0
+	for(var/obj/item/weapon/stock_parts/SP in component_parts)
+		if(istype(SP, /obj/item/weapon/stock_parts/manipulator))
+			manipCount += SP.rating
+	manipRating = manipCount
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/findConnections() //Finds and links the hub with the pods on either side
+	connectOne = findConnectOne()
+	if(connectOne)
+		connectOne.connectedHub = src
+		connectOne.podNumber = 1
+	connectTwo = findConnectTwo()
+	if(connectTwo)
+		connectTwo.connectedHub = src
+		connectTwo.podNumber = 2
+	if ((connectOne) && (connectTwo))
+		podsConnected = TRUE
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/findConnectOne()
+	for(var/obj/machinery/mind_machine/mind_machine_pod/CO in orange(1,src))
+		if(CO != connectTwo)
+			return CO
+/obj/machinery/mind_machine/mind_machine_hub/proc/findConnectTwo()
+	for(var/obj/machinery/mind_machine/mind_machine_pod/CT in orange(1,src))
+		if(CT != connectOne)
+			return CT
+
+/obj/machinery/mind_machine/mind_machine_hub/Destroy()
+	podsConnected = null
+	currentlySwapping = null
+	lockedPods = FALSE
+	occupantScan = FALSE
+	occupantOne = null
+	occupantTwo = null
+	swapProgress = 0
+	if(connectOne || connectTwo || podsConnected)
+		connectOne.connectedHub = null
+		connectTwo.connectedHub = null
+		connectOne.podNumber = 0
+		connectTwo.podNumber = 0
+		connectOne = null
+		connectTwo = null
+	. = ..()
+
+/obj/machinery/mind_machine/mind_machine_hub/attackby(var/obj/item/A as obj, var/mob/user as mob)
+	..()
+	if(istype(A, /obj/item/bluespace_crystal/flawless))
+		user.drop_item(A, src)
+		bluespaceConduit += 300
+		to_chat(user, "The bluespace conduit flashes violently before calming to a glow.")
+		qdel(A)
+		return
+	if(istype(A, /obj/item/bluespace_crystal/artificial))
```

Comment:
you have 3 crystals doing the same user.drop_item(A, src)(that needs to be an if() check because the player might not be able to drop the thing), you could simply have a single if for this. checking if it's a flawless/normal/artificial can be done inside that if